### PR TITLE
[libkqueue] new port

### PR DESCRIPTION
**Pull request by qmfrederik**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libkqueue/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libkqueue",
+  "version-semver": "2.6.2",
+  "description": "A user space implementation of the kqueue(2) kernel event notification mechanism libkqueue acts as a translator between the kevent structure and the native kernel facilities on Linux, Android, Solaris, and Windows.",
+  "homepage": "https://github.com/mheily/libkqueue/",
+  "license": "0BSD",
+  "supports": "windows & !uwp",
```
Comment (dg0yt) [inline ports/libkqueue/vcpkg.json:7]: The description promises much more than `windows & !uwp`.

File: ports/libkqueue/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    file(REMOVE "${CURRENT_PACKAGES_DIR}/lib/kqueue_static.lib")
+    file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/lib/kqueue_static.lib")
+endif()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+vcpkg_fixup_pkgconfig()
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```
Comment (dg0yt) [inline ports/libkqueue/portfile.cmake:43]: A line break at the end of this line is desired.

File: ports/libkqueue/vcpkg.json (hunk 1 above)
Comment (qmfrederik) [reply ports/libkqueue/vcpkg.json]: Fair enough, I updated the description.

File: ports/libkqueue/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    file(RENAME "${CURRENT_PACKAGES_DIR}/lib/kqueue_static.lib" "${CURRENT_PACKAGES_DIR}/lib/kqueue.lib")
+
+    file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/lib/kqueue.lib")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/kqueue_static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/kqueue.lib")
+    
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+else()
+    file(REMOVE "${CURRENT_PACKAGES_DIR}/lib/kqueue_static.lib")
+    file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/lib/kqueue_static.lib")
+endif()
```
[the remark this answers is not in the document]
Comment (qmfrederik) [reply ports/libkqueue/portfile.cmake]: So the upstream CMake file defines two targets, `kqueue` and `kqueue_static`.  If I were to patch the upstream CMakeLists, I would probably use something akin to [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/guide/tutorial/Selecting%20Static%20or%20Shared%20Libraries.html) to have the project output either a shared library or a static library, but never both.

Is that the approach you would take? It feels like a bit of 'deep surgery', and we risk having to update the patch whenever a new version is released?

I updated the portfile to support release-only triplets, but let me know what your preference is.

File: ports/libkqueue/portfile.cmake (hunk 2 above)
Comment (qmfrederik) [reply ports/libkqueue/portfile.cmake]: Done, thanks.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libkqueue/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libkqueue",
+  "version-semver": "2.6.2",
+  "description": "A user space implementation of the kqueue(2) kernel event notification mechanism libkqueue acts as a translator between the kevent structure and the native kernel facilities on Windows.",
+  "homepage": "https://github.com/mheily/libkqueue/",
+  "license": "0BSD",
+  "supports": "windows & !uwp",
```
Comment (dg0yt) [inline ports/libkqueue/vcpkg.json:7]: Is there no way add this port with all platforms suppored by upstream?
It is bad UX to find a port in vcpkg only to learn that it is incomplete.

File: ports/libkqueue/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libkqueue",
+  "version-semver": "2.6.2",
+  "description": "A user space implementation of the kqueue(2) kernel event notification mechanism libkqueue acts as a translator between the kevent structure and the native kernel facilities on Windows.",
```
Comment (dg0yt) [inline ports/libkqueue/vcpkg.json:4]: I didn't want Linux, Android, Solaris to be removed here. Remove all or nothing.

**Comment by JonLiu1993** [comment]
@qmfrederik, could you please response @dg0yt's review question? thanks.

**Comment by JonLiu1993** [comment]
Ping @qmfrederik for response.

**Comment by JonLiu1993** [comment]
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by qmfrederik** [comment]
Closing for now, will revisit later if time allows.

**Pull request closed without merging**